pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -217,6 +222,7 @@ func (r *NodeObservabilityReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&operatorv1alpha1.NodeObservability{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&corev1.Service{}).
+		WithEventFilter(reconcileClusterNamedResource()).
```

Comment:
This will filter all the watched objects, not only `NodeObservability`. To do what you wanted to you need to add this predicate as `ForOption` to the `For` function above.